ocrd_utils/ocrd_utils/str.py

```diff
@@ -177,3 +178,17 @@ def safe_filename(url):
     ret = re.sub('[^A-Za-z0-9]+', '.', url)
     #  print('safe filename: %s -> %s' % (url, ret))
     return ret
+
+def generate_range(start, end):
+    """
+    Generate a list of strings by incrementing the number part of ``start`` until including ``end``.
+    """
+    ret = []
+    start_num, end_num = re.search('\d+', start), re.search('\d+', end)
+    if not (start_num and end_num):
+        getLogger('ocrd_utils.generate_range').error("Unable to generate range %s .. %s, could not detect number part" % (start, end))
```

Comment:
Just a thought: Maybe this should even be an exception. After all, it's likely the entire result is not what was intended afterwards. Better safe than sorry.

Reply:
Fine with me.